htdocs/core/class/commonobject.class.php

```diff
@@ -6016,7 +6016,7 @@ function showOptionals($extrafields, $mode='view', $params=null, $keysuffix='',
 					// Convert date into timestamp format (value in memory must be a timestamp)
 					if (in_array($extrafields->attribute_type[$key],array('date','datetime')))
 					{
-						$value = GETPOSTISSET($keyprefix.'options_'.$key.$keysuffix)?dol_mktime(GETPOST($keyprefix.'options_'.$key.$keysuffix."hour",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."min",'int',3), 0, GETPOST($keyprefix.'options_'.$key.$keysuffix."month",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."day",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."year",'int',3)):$this->db->jdate($this->array_options['options_'.$key]);
+						$value = GETPOSTISSET($keyprefix.'options_'.$key.$keysuffix)?dol_mktime(GETPOST($keyprefix.'options_'.$key.$keysuffix."hour",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."min",'int',3), 0, GETPOST($keyprefix.'options_'.$key.$keysuffix."month",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."day",'int',3), GETPOST($keyprefix.'options_'.$key.$keysuffix."year",'int',3)):$this->db->idate($this->array_options['options_'.$key]);
```

Comment:
This change looks strange.
$value must be a timestamp so must be an integer value, so we must use jdate to convert string to timestamp.
Using idate make this oposite.

Formatting data $value into human readable date should be done later into function showOutputField

Reply:
This is the showOptionals function used to **display** extrafields.

$extrafields->showOutputField($key, $value) is used after but do not do any conversion based on type. (type is not a parameter of this function.

By the way I found the issue in a module I have developed and fixed it on my module by making the change.

I also test the resource module with extrafields but this module is just blowing away (no extra fields display in list and some strange issues with fields).  So it seems that calling extrafields functions and using TPL require a deep review for this module, all previous calls are not compatible.

Reply:
By the way it seems date extrafields in third party object are working fine.

 So I suspect the code is not implemented following the template you show me when I started to develop this module (it was Dolibarr 5 I think).

Reply:
I should also mention that in database my module date extrafields are stored as timestamp. So converting from timestamp to string is fine.

Reply:
OK. So i close the ticket and just reopen or create a new one if required.